lib/internal/modules/esm/hooks.js

```diff
@@ -553,15 +559,18 @@ class HooksProxy {
     }
   }
 
-  async makeAsyncRequest(method, ...args) {
+  async makeAsyncRequest(transferList, method, ...args) {
     this.waitForWorker();
 
     MessageChannel ??= require('internal/worker/io').MessageChannel;
     const asyncCommChannel = new MessageChannel();
 
     // Pass work to the worker.
-    debug('post async message to worker', { method, args });
-    this.#worker.postMessage({ method, args, port: asyncCommChannel.port2 }, [asyncCommChannel.port2]);
+    debug('post async message to worker', { method, args, transferList });
+    const finalTransferList = transferList ?
+      [asyncCommChannel.port2, ...transferList] :
+      [asyncCommChannel.port2]
+    this.#worker.postMessage({ method, args, port: asyncCommChannel.port2 }, finalTransferList);
```

Comment:
```suggestion
    this.#worker.postMessage({ __proto__: null, method, args, port: asyncCommChannel.port2 }, finalTransferList);
```
altho this might not matter if the worker's not in object mode and if serialization ignores non-own properties